src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-modal/osd-devices-selection-modal.component.ts

```diff
@@ -54,7 +54,10 @@ export class OsdDevicesSelectionModalComponent implements AfterViewInit {
     const cols = _.filter(this.inventoryDevices.columns, (col) => {
       return this.filterColumns.includes(col.prop) && col.prop !== 'hostname';
     });
-    this.requiredFilters = _.map(cols, 'name');
+    // Fixes 'ExpressionChangedAfterItHasBeenCheckedError'
+    setTimeout(() => {
+      this.requiredFilters = _.map(cols, 'name');
+    }, 0);
```

Comment:
Thanks for fixing the issue.
If this hack is needed, I wonder if the design was right when I implemented this...

Reply:
It's now needed because of the modal change. It seems like the new modal initializes the modal somehow in one step with all other components used at the same time (which of course is faster but has downsides). Through the new behavior will set `requieredFilters` multiple times in one run, which causes the error. The timeout just adds an extra "frame" that forces the template to update after all that is triggered is executed. This way we don't change the variable more than once in a template run.

Before that change your implementation worked fine :+1: